components/user/EventRow.tsx

```diff
@@ -0,0 +1,296 @@
+}
+export type EventRowProps = {
+  id: number
+  type: EventType
+  occurred_at: string
+  points: number
+  metadata?: ApiEventMetadata
+}
+
+const makeLinkForEvent = (type: EventType, metadata?: ApiEventMetadata) => {
+  if (!metadata) return ''
+  if ('hash' in metadata && type === EventType.BLOCK_MINED) {
+    return `https://explorer.ironfish.network/blocks/${metadata.hash}`
+  }
+  return (metadata as ApiEventMetadataWithLink).url
+}
+
+type CopyableHashProps = {
+  hash: string
+}
+
+enum CopyState {
+  IDLE = 'copy_idle',
+  OVER = 'copy_over',
+  COPIED = 'copy_copied',
+  RESETTING = 'copy_resetting',
+}
+
+const CopyableHash = ({ hash }: CopyableHashProps) => {
+  const [, $setCopied] = useClipboard(hash, {
+    successDuration: 1000,
+  })
+  const [$hover, $setHover] = useState<CopyState>(CopyState.IDLE)
+  const abbrevHash = hash.slice(45, Infinity)
+  return (
+    <div
+      title={hash}
+      onClick={e => {
+        e.preventDefault()
+        $setCopied()
+        $setHover(CopyState.COPIED)
+        setTimeout(() => $setHover(CopyState.RESETTING), 2e3)
+      }}
+      onMouseOver={() => {
+        if ($hover === CopyState.COPIED) {
+          return
+        }
+        $setHover(CopyState.OVER)
+      }}
+      onMouseOut={() => {
+        if ($hover !== CopyState.COPIED && $hover !== CopyState.RESETTING) {
+          $setHover(CopyState.IDLE)
+        }
+      }}
+      className={clsx(
+        'flex',
+        'items-center',
+        'justify-center',
+        'relative',
+        styles.copyable,
+        $hover === CopyState.RESETTING
+          ? styles.resetting
+          : $hover === CopyState.COPIED
+          ? styles.copied
+          : $hover === CopyState.OVER
+          ? styles.active
+          : ''
+      )}
+    >
+      <>
+        Block &hellip; {abbrevHash}
+        <ActivityCopy className="ml-4" />
+      </>
+    </div>
+  )
+}
+
+const summarizeEvent = (
+  type: EventType,
+  metadata: ApiEventMetadata
+): ReactElement => {
+  if (type === EventType.BLOCK_MINED) {
+    // return 'View in the explorer'
+    const { hash } = metadata as ApiEventMetadataBlockMined
+    return <CopyableHash hash={hash} />
+  }
+  const { url } = metadata as ApiEventMetadataWithLink
+  const link = new URL(url)
+  const { pathname, hostname } = link
+  const parts = pathname.split('/')
+  const id = parts.slice(-1)[0]
+  if (type === EventType.PULL_REQUEST_MERGED) {
+    // https://github.com/iron-fish/ironfish-api/pull/595
+    return <>View pull request #{id}</>
+  } else if (type === EventType.BUG_CAUGHT) {
+    // https://github.com/iron-fish/ironfish/issues/930
+    return <>View issue #{id}</>
+  } else if (type === EventType.COMMUNITY_CONTRIBUTION) {
+    return <>View contribution</>
+  } else if (type === EventType.SOCIAL_MEDIA_PROMOTION) {
+    return <>Promoted on {hostname}</>
+  }
+  return <>UNHANDLED: {type}</>
+}
+
+const formatEventDate = (d: Date) =>
+  format(d, `do MMM. Y - kk':'mm':'ss`, {
```

Comment:
This format style is a bit different from what's in the Linear issue, did someone decide to change it?

Reply:
Reverted